dal/UserDal.js

```diff
@@ -0,0 +1,83 @@
+const Dal = require('./Dal');
+const User = require('../models/User');
+const SessionService = require('../services/Session');
+
+class UserDal extends Dal {
+    constructor() {
+        super();
+    }
+
+    async createUser(user) {
+        try {
+            await this.mongo.connect();
+            await user.save();
+            return null; // We don't need to return the user details.
+        } catch (err) {
+            if (err.code === 11000) { // Duplicated Unique Key
+                throw new this.RequestError(`Duplicated Key: ${JSON.stringify(err.keyValue)}`);
+            }
+            console.log(err);
+            throw err;
+        }
+    }
+
+    async login(username) {
+        try {
+            await this.mongo.connect();
+            return await User.findOne({username});
+        } catch (err) {
```

Comment:
IMO this is redundant, you are not doing anything with the exception. The runtime itself will print it.